File: 백준/Bronze/2798. 블랙잭/블랙잭.py
n, m = map(int, input().split())
card = list(map(int, input().split()))

card.sort()

result = 0
for i in range(n-2):
    if card[i] > m:
        break
    for j in range(i + 1, n-1):
        for k in range(j + 1, n):
            if card[i] + card[j] + card[k] > m:
                continue
            result = max(result, card[i] + card[j] + card[k])
            
print(result)

File: 백준/Bronze/2798. 블랙잭/블랙잭.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String[] tokens = sc.nextLine().split(" ");
		int n = Integer.parseInt(tokens[0]);
		int m = Integer.parseInt(tokens[1]);

		String[] line = sc.nextLine().split(" ");
		int[] arr = new int[line.length];
		for (int i = 0; i < line.length; i++) {
			arr[i] = Integer.parseInt(line[i]);
		}
		Arrays.sort(arr);

		int ans = 0;

		for (int i = 0; i < n - 2; i++) {
			if (arr[i] > m)
				break;
			for (int j = i + 1; j < n - 1; j++) {
				for (int k = j + 1; k < n; k++) {
					if ((arr[i] + arr[j] + arr[k]) > m) {
						continue;
					}
					ans = Math.max(ans, arr[i] + arr[j] + arr[k]);
				}
			}
		}
		System.out.println(ans);
	}
}
